Core/src/Geometry/CuboidVolumeStack.cpp

```diff
@@ -0,0 +1,893 @@
+    return;
+  }
+  for (const auto& vt : volumes) {
+    std::stringstream ss;
+    ss << std::fixed;
+    ss << std::setprecision(3);
+    ss << std::setfill(' ');
+
+    int w = 9;
+
+    for (const auto& axis :
+         {AxisDirection::AxisX, AxisDirection::AxisY, AxisDirection::AxisZ}) {
+      ss << axisDirectionName(axis) << ": [ " << std::setw(w) << vt.min(axis)
+         << " <- " << std::setw(w) << vt.mid(axis) << " -> " << std::setw(w)
+         << vt.max(axis) << " ]\n";
+    }
+    logger().log(lvl, ss.str());
+  }
+}
+
+void CuboidVolumeStack::checkVolumeAlignment(
+    const std::vector<VolumeTuple>& volumes, const Logger& logger) const {
+  std::size_t n = 0;
+  auto dirIdx = static_cast<std::size_t>(m_dir);
+  auto dirOrth1Idx = static_cast<std::size_t>(m_dirOrth1);
+  auto dirOrth2Idx = static_cast<std::size_t>(m_dirOrth2);
+
+  for (auto& vt : volumes) {
+    ACTS_VERBOSE("Checking volume #"
+                 << n << " at " << axisDirectionName(m_dir) << ": "
+                 << vt.localTransform.translation()[dirIdx]);
+    ACTS_VERBOSE("- Local transform is:\n" << vt.localTransform.matrix());
+
+    // @TODO: Rotation precision?
+    constexpr auto tolerance = s_onSurfaceTolerance;
+
+    // In the group coordinate system:
+
+    // a) the volumes cannot have any relative rotation
+    if ((vt.localTransform.rotation().matrix() - RotationMatrix3::Identity())
+            .norm() > tolerance) {
+      ACTS_ERROR("Volumes are not aligned: rotation is different");
+      throw std::invalid_argument(
+          "Volumes are not aligned: rotation is different");
+    }
+
+    ACTS_VERBOSE(" -> Rotation is ok!");
+
+    // b) the volumes cannot have translation in orthogonal directions
+    if (std::abs(vt.localTransform.translation()[dirOrth1Idx]) > tolerance ||
+        std::abs(vt.localTransform.translation()[dirOrth2Idx]) > tolerance) {
+      ACTS_ERROR("Volumes are not aligned: translation in "
+                 << axisDirectionName(m_dirOrth1) << " or "
+                 << axisDirectionName(m_dirOrth2));
+      throw std::invalid_argument("Volumes are not aligned: translation in " +
+                                  axisDirectionName(m_dirOrth1) + " or " +
+                                  axisDirectionName(m_dirOrth2));
+    }
+    ACTS_VERBOSE(" -> Translation in " << axisDirectionName(m_dirOrth1) << "/"
+                                       << axisDirectionName(m_dirOrth2)
+                                       << " is ok!");
+
+    n++;
+  }
+}
+
+std::pair<double, double> CuboidVolumeStack::synchronizeBounds(
+    std::vector<VolumeTuple>& volumes, const Logger& logger) {
+  auto boundDirOrth1 = static_cast<CuboidVolumeBounds::BoundValues>(m_dirOrth1);
+  auto boundDirOrth2 = static_cast<CuboidVolumeBounds::BoundValues>(m_dirOrth2);
+
+  const double maxHl1 =
+      std::max_element(volumes.begin(), volumes.end(),
+                       [&boundDirOrth1](const auto& a, const auto& b) {
```

Comment:
```suggestion
                       [boundDirOrth1](const auto& a, const auto& b) {
```

Not a huge difference, but the `&` is not necessary here, because it's not mutated, and the size of the integer being captured is the same as the pointer.